lua/ulx/modules/sh/cfc_propify.lua

```diff
@@ -0,0 +1,219 @@
+propifyCommand:addParam{ type = ULib.cmds.StringArg, default = PROP_DEFAULT_MODEL, ULib.cmds.optional }
+propifyCommand:addParam{ type = ULib.cmds.BoolArg, invisible = true }
+propifyCommand:defaultAccess( ULib.ACCESS_ADMIN )
+propifyCommand:help( "Turns the target(s) into a prop with the given model." )
+propifyCommand:setOpposite( "ulx unpropify", { _, _, _, true }, "!unpropify" )
+
+local function propifySpawnCheck( ply )
+    if ply.ragdoll then
+        timer.Simple( 0.01, function()
+            if not ply:IsValid() then return end
+            ply:Spectate( OBS_MODE_CHASE )
+            ply:SpectateEntity( ply.ragdoll )
+            ply:StripWeapons()
+        end )
+    end
+end
+hook.Add( "PlayerSpawn", "CFC_ULXPropifySpawnCheck", propifySpawnCheck )
+
+local function propDisconnectedCheck( ply )
+    if ply.ragdoll then
+        ply.ragdoll:DisallowDeleting( false )
+        ply.ragdoll:Remove()
+    end
+end
+hook.Add( "PlayerDisconnected", "CFC_ULXPropDisconnectedCheck", propDisconnectedCheck, HOOK_MONITOR_HIGH )
+
+local function removePropOnCleanup()
+    local players = player.GetAll()
+    for _, v in pairs( players ) do
+        if v.ragdoll then
+            v.propifyAfterCleanup = true
+            unpropifyPlayer( v )
+        end
+    end
+end
+hook.Add( "PreCleanupMap", "CFC_ULXPropBeforeCleanup", removePropOnCleanup )
+
+local function createPropAfterCleanup()
+    local players = player.GetAll()
+    for _, v in pairs( players ) do
+        if v.propifyAfterCleanup then
+            v.propifyAfterCleanup = nil
+            timer.Simple( 0.1, function() propifyPlayer( v ) end )
+        end
+    end
+end
+hook.Add( "PostCleanupMap", "CFC_ULXPropAfterCleanup", createPropAfterCleanup )
+
+--Player movement:
+local function propHop( ply, keyNum )
+    if not ply.ragdoll or not ply.ragdoll:GetClass() == "prop_physics" then return end
+    
+    ply.propifiedlastPressed = ply.propifiedlastPressed or 0
+    
+    if ply.propifiedlastPressed + HOP_COOLDOWN > CurTime() then return end
+    
+    ply.propifiedlastPressed = CurTime()
+    
+    if keyNum == IN_FORWARD then
+        local phys = ply.ragdoll:GetPhysicsObject()
+        phys:ApplyForceCenter( ply:EyeAngles():Forward() * HOP_STRENGTH * phys:GetMass() )
+    elseif keyNum == IN_BACK then
+        local phys = ply.ragdoll:GetPhysicsObject()
+        phys:ApplyForceCenter( -ply:EyeAngles():Forward() * HOP_STRENGTH * phys:GetMass() )
+    elseif keyNum == IN_MOVERIGHT then
+        local phys = ply.ragdoll:GetPhysicsObject()
+        phys:ApplyForceCenter( ply:EyeAngles():Right() * HOP_STRENGTH * phys:GetMass() )
+    elseif keyNum == IN_MOVELEFT then
+        local phys = ply.ragdoll:GetPhysicsObject()
+        phys:ApplyForceCenter( -ply:EyeAngles():Right() * HOP_STRENGTH * phys:GetMass() )
+    elseif keyNum == IN_JUMP then
+        local phys = ply.ragdoll:GetPhysicsObject()
+        phys:ApplyForceCenter( Vector( 0, 0, HOP_STRENGTH * phys:GetMass() ) )
+    end
```

Comment:
- set `ply:EyeAngles()` to a local variable
- set `HOP_STRENGTH * phys:GetMass()` as a local variable as well
- phys should just be defined once before the if elseif chain

so it would become something like
```lua
if keyNum == IN_FORWARD then
    phys:ApplyForceCenter( eyeAngles:Forward() * hopStrength )
elseif ... etc
```